components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogQuery.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Infrastructure\Persistence\Catalog;
+
+use Akeneo\Catalogs\Domain\Catalog\Catalog;
+use Akeneo\Catalogs\Domain\Catalog\GetCatalogQueryInterface;
+use Doctrine\DBAL\Connection;
+use Ramsey\Uuid\Uuid;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @phpstan-type DatabaseCatalog array{
+ *      id: string,
+ *      name: string,
+ *      owner_username: string,
+ *      is_enabled: string,
+ *      product_selection_criteria: string,
+ *      product_value_filters: string,
+ * }
+ *
+ * @phpstan-import-type ProductSelectionCriterion from Catalog
+ * @phpstan-import-type ProductValueFilters from Catalog
+ */
+final class GetCatalogQuery implements GetCatalogQueryInterface
+{
+    public function __construct(private Connection $connection)
+    {
+    }
+
+    public function execute(string $id): Catalog
```

Comment:
```suggestion
    public function execute(string $catalogId): Catalog
```
As it's not detailed in the query name `GetCatalogById` I would have named the argument $catalogId, because there is an owner id too in the aggregate.